src/domain/flow-system/services/src/flow/flow_permissions_plugin_impl.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use dill::*;
+use futures::{StreamExt, TryStreamExt};
+use kamu_accounts::AuthenticationService;
+use kamu_core::{DatasetRepository, DependencyGraphService, InternalError};
+use kamu_flow_system::*;
+use opendatafabric::{AccountID, DatasetID};
+
+pub struct FlowPermissionsPluginImpl {
+    dataset_repo: Arc<dyn DatasetRepository>,
+    authentication_svc: Arc<dyn AuthenticationService>,
+    dependency_graph_service: Arc<dyn DependencyGraphService>,
+}
+
+#[component(pub)]
+#[interface(dyn FlowPermissionsPlugin)]
+#[scope(Singleton)]
+impl FlowPermissionsPluginImpl {
+    pub fn new(
+        dataset_repo: Arc<dyn DatasetRepository>,
+        authentication_svc: Arc<dyn AuthenticationService>,
+        dependency_graph_service: Arc<dyn DependencyGraphService>,
+    ) -> Self {
+        Self {
+            dataset_repo,
+            authentication_svc,
+            dependency_graph_service,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl FlowPermissionsPlugin for FlowPermissionsPluginImpl {
+    async fn get_account_downstream_dependencies(
+        &self,
+        dataset_id: &DatasetID,
+        account_id_maybe: Option<AccountID>,
+    ) -> Result<Vec<DatasetID>, InternalError> {
```

Comment:
This thing is neither specifict to flow system in trait definition, nor in implementation.
I think with some more generic name, it should live in the core.